src/engines/brave.rs

```diff
@@ -0,0 +1,95 @@
+//! The `brave` module handles the scraping of results from the brave search engine
+//! by querying the upstream brave search engine with user provided query and with a page
+//! number if provided.
+
+use std::collections::HashMap;
+
+use reqwest::header::HeaderMap;
+use scraper::Html;
+
+use crate::models::aggregation_models::SearchResult;
+use error_stack::{Report, Result, ResultExt};
+
+use crate::models::engine_models::{EngineError, SearchEngine};
+
+use super::search_result_parser::SearchResultParser;
+
+/// Scrapes the results from the Brave search engine.  
+pub struct Brave {
+    /// Utilises generic logic for parsing search results.
+    parser: SearchResultParser,
+}
+
+impl Brave {
+    /// Creates the Brave parser.
+    pub fn new() -> Result<Brave, EngineError> {
+        Ok(Self {
+            parser: SearchResultParser::new(
+                "#results h4",
+                "#results [data-pos]",
+                "a > .url",
+                "a",
+                ".snippet-description",
+            )?,
+        })
+    }
+}
+
+#[async_trait::async_trait]
+impl SearchEngine for Brave {
+    async fn results(
+        &self,
+        query: &str,
+        page: u32,
+        user_agent: &str,
+        request_timeout: u8,
+        safe_search: u8,
+    ) -> Result<HashMap<String, SearchResult>, EngineError> {
+        let url = format!("https://search.brave.com/search?q={query}&offset={page}");
+
+        let safe_search_level = match safe_search {
+            0 => "off",
+            1 | 2 => "moderate",
```

Comment:
```suggestion
            1  => "moderate",
```

Ok I would suggest a final minor change, Actually the strict mode should be enabled for any `safe_search_level` value above 1. So I would recommend making this change and after this we will merge the PR. :slightly_smiling_face: